test_mocks/list-unix.js

```diff
@@ -0,0 +1,53 @@
+// Mocks fs.read for listUnix
+
+'use strict';
+
+var SandboxedModule = require('sandboxed-module');
+
+var mockPorts = {};
+var error = false;
+
+var listUnix = SandboxedModule.require('../lib/list-unix', {
+  requires: {
+    fs: {
+      readdir: function(path, cb) {
+        if (error) {
+          return process.nextTick(cb.bind(null, new Error('bad')));
+        }
+        process.nextTick(cb.bind(null, null, Object.keys(mockPorts)));
+      },
+      statSync: function() {
+        return { isCharacterDevice: function() { return true } };
+      }
+    },
+    path: {
+      // needed for testing on windows
+      join: function() {
+        return Array.prototype.join.call(arguments, '/');
+      }
+    },
+    child_process: {
+      exec: function(cmd, cb) {
+        var port = cmd.split(/\/dev\/(.*)\)/)[1];
+        process.nextTick(function() {
+          cb(null, mockPorts[port]);
+        });
+      }
+    }
+  }
```

Comment:
Disagree. 

- https://github.com/tessel/t2-cli/blob/master/test/unit/deploy.js#L1709-L1724
- https://github.com/tessel/t2-cli/blob/master/test/unit/deploy.js#L1837
- https://github.com/tessel/t2-cli/blob/master/test/unit/deploy.js#L1900-L1918
- https://github.com/tessel/t2-cli/blob/master/test/unit/deploy.js#L1934-L1936
- https://github.com/tessel/t2-cli/blob/master/test/unit/deploy.js#L1944

And tons more